build-system/tasks/dist.js

```diff
@@ -114,19 +114,20 @@ async function dist() {
   await runPreDistSteps(options);
 
   // These steps use closure compiler. Small ones before large (parallel) ones.
-  const steps = [];
   if (argv.core_runtime_only) {
-    steps.push(compileCoreRuntime(options));
+    await compileCoreRuntime(options);
   } else {
-    steps.push(buildExperiments());
-    steps.push(buildLoginDone('0.1'));
-    steps.push(buildWebPushPublisherFiles());
-    steps.push(buildCompiler());
-    steps.push(compileAllJs(options));
+    await Promise.all([
+      buildExperiments(),
+      buildLoginDone('0.1'),
+      buildWebPushPublisherFiles(),
+      buildCompiler(),
+    ]);
+    await compileAllJs(options);
```

Comment:
I think you can also include this in the `Promise.all`. I believe its only the extensions and vendor configs that already use `esbuild`

Reply:
Trying this with 100 VMs: https://app.circleci.com/pipelines/github/ampproject/amphtml/17036/workflows/e9cc67ec-554a-4535-91d3-c16087342494

Reply:
Change looks safe. Moved it to this PR!